domains/authorization/Authorization.API/API/ValueObjects/ClientName.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace API.ValueObjects;
+
+public class ClientName : ValueObject
+{
+    public string Value { get; }
+
+    public ClientName(string value)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+            throw new ArgumentException("Value cannot be null, empty, or whitespace.", nameof(value));
+
+        if (!IsValidClientName(value))
+            throw new ArgumentException("Value contains invalid characters.", nameof(value));
+
+        Value = value.Trim();
+    }
+
+    private static bool IsValidClientName(string value) => MyRegex().IsMatch(value);
+
+    private static Regex MyRegex() => new Regex(@"^[a-zA-Z0-9\s&.,'\-]+$");
```

Comment:
Nit: This regex doesn't allow slashes. Can companies have slashes in their names?

Reply:
fixed